lib/rom/sql/attribute.rb

```diff
@@ -14,6 +14,49 @@ class Attribute < ROM::Schema::Attribute
       # Error raised when an attribute cannot be qualified
       QualifyError = Class.new(StandardError)
 
+      # Type-specific methods
+      #
+      # @api public
+      module TypeExtensions
+        class << self
+          # Gets extensions for a type
+          #
+          # @param [Dry::Types::Type] type
+          #
+          # @return [Hash]
+          #
+          # @api public
+          def [](type)
+            unwrapped = type.optional? ? type.right : type
```

Comment:
I wonder if it's possible to have a case where `type.right` would be...another optional *heh*